app/web/features/userQueries/useLiteUsers.test.tsx

```diff
@@ -0,0 +1,283 @@
+
+    it("Uses cached data instead of refetching", async () => {
+      const { result } = renderHook(() => useLiteUsers(ids), {
+        wrapper: sharedClientWrapper,
+      });
+
+      expect(mockGetLiteUsers).not.toHaveBeenCalled();
+
+      expect(result.current.isFetching).toBe(false);
+      expect(result.current.isLoading).toBe(false);
+    });
+
+    it("Should return cached data and set isError when fetching users fails", async () => {
+      mockConsoleError();
+      mockGetLiteUsers.mockRejectedValue(
+        new RpcError(StatusCode.INTERNAL, "Error fetching user data", {})
+      );
+
+      const { result, waitForNextUpdate } = renderHook(
+        () => useLiteUsers([1, 2, 3]),
+        { wrapper: sharedClientWrapper }
+      );
+
+      await waitForNextUpdate();
+
+      expect(result.current).toEqual(
+        expect.objectContaining({
+          data: undefined,
+          error: new RpcError(
+            StatusCode.INTERNAL,
+            "Error fetching user data",
+            {}
+          ),
+          isError: true,
+          isFetching: false,
+          isLoading: false,
+          isRefetching: false,
+        })
+      );
+    });
+  });
+
+  describe("useLiteUser (singular)", () => {
+    const mockGetLiteUser = service.user.getLiteUser as jest.Mock;
+
+    it("Should return loading state correctly", async () => {
+      mockGetLiteUser.mockResolvedValue(getLiteUser("1"));
+
+      const { result, waitFor } = renderHook(() => useLiteUser(1), { wrapper });
+
+      expect(result.current.isLoading).toBe(true);
+      expect(result.current.data).toBeUndefined();
+
+      await waitFor(() => expect(result.current.isLoading).toBe(false));
+    });
+
+    it("Should return user when loading is complete", async () => {
+      mockGetLiteUser.mockResolvedValue(getLiteUser("2"));
+
+      const { result, waitForNextUpdate } = renderHook(() => useLiteUser(1), {
+        wrapper,
+      });
+
+      await waitForNextUpdate();
+      expect(mockGetLiteUser).toHaveBeenCalledTimes(1);
+
+      expect(result.current).toEqual(
+        expect.objectContaining({
+          data: mockLiteUsers[1],
+          error: null,
+          isError: false,
+          isFetching: false,
+          isLoading: false,
+          isRefetching: false,
+        })
+      );
+    });
+
+    it("Should return an error if the query fails", async () => {
+      mockConsoleError();
+      const error = new RpcError(500, "Some error", {});
+      mockGetLiteUser.mockRejectedValue(error);
+
+      const { result, waitForNextUpdate } = renderHook(() => useLiteUser(1), {
+        wrapper,
+      });
+
+      await waitForNextUpdate();
+
+      expect(result.current.isLoading).toBe(false);
+      expect(result.current.error).toBe(error);
+      expect(result.current.data).toBeUndefined();
+    });
+
+    it("Should not run the query function when id is undefined", () => {
+      // @ts-expect-error
+      renderHook(() => useLiteUser(), { wrapper });
```

Comment:
```suggestion
      renderHook(() => useLiteUser(undefined), { wrapper });
```
A bit more verbose but then you won't need the `ts-expect-error` comment? I think it's because it's defined with `| undefined` instead of `?:` in the type annotation, so it becomes a required argument even when the value is `undefined`

Reply:
Yes, you are correct! Fixed!